Homework7/Calculator/FiniteStateMachine.cs

```diff
@@ -0,0 +1,122 @@
+﻿namespace CalculatorSpace;
+
+internal class FiniteStateMachine
+{
+    enum States : int
+    {
+        firstOperandStartState = 0,
+        firstOperandState = 1,
+        signState = 2,
+        plusOperatorState = 3,
+        secondOperandStartState = 4,
+        secondOperandState = 5,
+    }
+
+    private States currentState = States.firstOperandStartState;
+    private string firstOperand = string.Empty;
+    private string secondOperand = string.Empty;
+    static char Operator;
```

Comment:
```suggestion
    private static char Operator;
```
И вообще, статические поля нельзя, это глобальные переменные